app/authentication/server/startup/index.js

```diff
@@ -363,8 +363,17 @@ Accounts.validateNewUser(function(user) {
 		return true;
 	}
 
-	if (settings.get('Accounts_Registration_AuthenticationServices_Enabled') === false && settings.get('LDAP_Enable') === false && !(user.services && user.services.password)) {
-		throw new Meteor.Error('registration-disabled-authentication-services', 'User registration is disabled for authentication services');
+	if (settings.get('Accounts_Registration_AuthenticationServices_Enabled') === false) {
+		if (settings.get('LDAP_Enable') === false && !(user.services && user.services.password)) {
+			throw new Meteor.Error('registration-disabled-authentication-services', 'User registration is disabled for authentication services');
```

Comment:
I split the original check into two parts, the `settings.get('Accounts_Registration_AuthenticationServices_Enabled') === false` and `settings.get('LDAP_Enable') === false && !(user.services && user.services.password)`, which doesn't change the logic